src/data-mgt/python/jobs/device_registry.py

```diff
@@ -0,0 +1,131 @@
+import json
+import os
+import requests
+
+AIRQO_API_BASE_URL = os.getenv("AIRQO_API_BASE_URL")
+
+MEASUREMENT_UNITS = dict({
+    "temperature": "°C",
+    "relHumid": "%",
+    "pm10ConcMass": "μg/m3",
+    "pm10ConcNum": "micrograms",
+    "pm1ConcMass": "μg/m3",
+    "pm1ConcNum": "micrograms",
+    "pm2_5ConcMass": "μg/m3",
+    "pm2_5ConcNum": "micrograms",
+    "no2Conc": "ppb",
+    "vocConc": "ppb",
+    "co2Conc": "ppb",
+})
+
+
+def single_component_insertion(data, tenant):
+
+    """
+    inserts a single device component data into the events table
+    :param data: component data (includes component name and device id to reduce on function args)
+    :param tenant: organisation eg airqo
+    :return: none
+    """
+
+    try:
+
+        # extract the component name and device id from the data
+        device = data.pop("device")
+
+        # create a json object of the remaining data and post to events table
+        json_data = json.dumps([data])
+        # print(json_data)
+        headers = {'Content-Type': 'application/json'}
+        url = AIRQO_API_BASE_URL + "devices/events/add?device=" + device + "&tenant=" + tenant
+
+        results = requests.post(url, json_data, headers=headers)
+
+        print(results.json())
+
+    except Exception as e:
+        print("================ Error Occurred ==================")
+        print(e)
+        print("================ Error End ==================")
+
+
+def get_component_type(component, tenant):
+
+    """
+    gets a component type if it exists, else creates it
+    :param component: component name
+    :param tenant: organisation eg airqo
+    :raises an exception if it fails to retrieve or register a component type
+    :return: component type
+    """
+
+    # get component type
+    existing_component_type_url = AIRQO_API_BASE_URL + "devices/list/components/types?name=" + component + "&tenant=" + tenant
+    existing_component_type_results = requests.get(existing_component_type_url)
+
+    # register the component type if it doesnt exists
+    if not existing_component_type_results.json()["doesExist"]:
+        get_component_type_url = AIRQO_API_BASE_URL + "devices/add/components/types?name=" + component + "&tenant=" + tenant
+        component_type_results = requests.post(get_component_type_url)
+
+        if not component_type_results.json()["doesExist"]:
+            # print(component_type_results.json())
+            raise Exception("failed to create and retrieve component type")
+
+        return component_type_results.json()["componentType"][0]["name"]
+
+    else:
+        return existing_component_type_results.json()["componentType"][0]["name"]
+
+
+def get_component_details(device_code, component, tenant):
```

Comment:
Same case, I am not sure if we need to get component details  since we are now device focused.